cmd/podman/cp.go

```diff
@@ -271,25 +261,8 @@ func parsePath(runtime *libpod.Runtime, path string) (*libpod.Container, string)
 	return nil, path
 }
 
-func getPathInfo(path string) (string, os.FileInfo, error) {
-	path, err := filepath.EvalSymlinks(path)
-	if err != nil {
-		return "", nil, errors.Wrapf(err, "error evaluating symlinks %q", path)
-	}
-	srcfi, err := os.Stat(path)
-	if err != nil {
-		return "", nil, errors.Wrapf(err, "error reading path %q", path)
-	}
-	return path, srcfi, nil
-}
-
-func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, chownOpts *idtools.IDPair, extract, isFromHostToCtr bool) error {
-	srcPath, err := filepath.EvalSymlinks(src)
-	if err != nil {
-		return errors.Wrapf(err, "error evaluating symlinks %q", srcPath)
-	}
-
-	srcPath, srcfi, err := getPathInfo(srcPath)
+func copy(srcPath, destPath, dest string, idMappingOpts storage.IDMappingOptions, chownOpts *idtools.IDPair, extract, isFromHostToCtr bool) error {
+	srcfi, err := os.Stat(srcPath)
```

Comment:
This re-evaluates symlinks.

Sure, `cleanPath` has evaluated them once, so this should “only” matter if the container is running and racing against this code — but the `Pause` option is off by default.

- Why is it off by default? That makes it basically pointless.
- Why is it even an option? We shouldn’t have “break my system” options.

Reply:
@mheon Can you answer @mtrmac please?

Reply:
Let me dig up the old PR where that fix happened - originally, pause was optional, but default-on. We made the determination that this was a bad idea (copying a multi-gigabyte file would keep a container paused for a few minutes on a slower system, for example - and that seemed bad). I think the consensus was to disable it but leave the option available for those who want to be safe.

Reply:
https://github.com/containers/libpod/pull/3214 merged with pause as default-true, but I know we changed that - still looking for where

Reply:
https://github.com/containers/libpod/pull/3279

Fixed here, with in an unrelated PR, with no comments as to why. Past-me was a bad open source citizen.

I think @mtrmac is probably write about avoiding the re-evaluation here, given that we do not guarantee that containers are paused.

Reply:
Well, there is `os.Lstat`, but AFAICS `copyFileWithTar` is using `Stat` and `os.Open` again, so, just fixing this one line isn’t enough anyway.